api/topo/topo.proto

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2019-present Open Networking Foundation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+// Package topo defines interfaces for managing network top
+package topo;
+
+import "google/protobuf/any.proto";
+import "gogoproto/gogo.proto";
+
+// EntityService provides an API for managing entities.
+service topo {
+  // Update one or more entities to the topology
+  rpc Write (WriteRequest) returns (WriteResponse) {
+  }
+
+  // Read one or more entities from topology
+  rpc Read (ReadRequest) returns (ReadResponse) {
+  }
+
+  // Represents the bidirectional stream between
+  // onos-topo and a client for the purpose of
+  // - streaming notifications
+  rpc StreamChannel (stream StreamMessageRequest) returns (stream StreamMessageResponse) {
```

Comment:
this is for CRUD operations I presume - should it be broken up in to these 4, or at least read and write?

Reply:
Read and Write rps are above. The Update message of the Write rpc is overloaded for Create, Update, Delete.

Reply:
i was referring to the StreamChannel - what is its purpose. Will there be any streaming reader? The GUI would benefit from it.

Reply:
Havent given the streaming channel much thought yet but the general idea is for a streaming reader (like GUI) to be able to subscribe to streaming topo events.